### PR TITLE
__noinline__ the top level igamma cuda kernel.

### PR DESCRIPTION
**Pull request by gchanan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47414 __noinline__ the top level igamma cuda kernel.**
* #47410 Move igamma cuda specific code to kernel file.

This improves the build time of this file by 10x on my machine (~12 minutes to ~1 minute).

Differential Revision: [D24746458](https://our.internmc.facebook.com/intern/diff/D24746458)

### CODE REVIEW

**Comment by gchanan** [comment]
Thanks to @mruberry and @ngimel for help with the fix.

File: aten/src/ATen/native/cuda/IGammaKernel.cu [hunk 1]
```diff
@@ -365,7 +365,7 @@ __host__ __device__ scalar_t _igamc_helper_continued_fraction(scalar_t a, scalar
 }
 
 template <typename scalar_t>
-inline __host__ __device__ scalar_t calc_igammac(scalar_t a, scalar_t x) {
+__host__ __device__ scalar_t calc_igammac(scalar_t a, scalar_t x) {
```
Comment (mruberry) [inline aten/src/ATen/native/cuda/IGammaKernel.cu:368]: Do you want to preemptively `__noinline__` this, too? It's about to be exposed directly to users.

**Review by mruberry — APPROVED** [review]
Great detective work!

File: aten/src/ATen/native/cuda/IGammaKernel.cu (hunk 1 above)
Comment (gchanan) [reply aten/src/ATen/native/cuda/IGammaKernel.cu:368]: will do

**Pull request closed without merging**